src/main/webapp/app/exercises/shared/exercise/exercise.service.ts

```diff
@@ -55,12 +55,13 @@ export class ExerciseService {
      */
     validateDate(exercise: Exercise) {
         exercise.dueDateError = exercise.releaseDate && exercise.dueDate ? !exercise.dueDate.isAfter(exercise.releaseDate) : false;
-
         exercise.assessmentDueDateError =
-            exercise.assessmentDueDate && exercise.releaseDate
-                ? !exercise.assessmentDueDate.isAfter(exercise.releaseDate)
-                : exercise.assessmentDueDate && exercise.dueDate
-                ? !exercise.assessmentDueDate.isAfter(exercise.dueDate)
+            exercise.releaseDate && exercise.assessmentDueDate
+                ? exercise.assessmentDueDate.isAfter(exercise.releaseDate)
+                    ? exercise.dueDate && exercise.assessmentDueDate && !exercise.assessmentDueDate.isAfter(exercise.dueDate)
+                    : true
+                : exercise.dueDate && exercise.assessmentDueDate
+                ? exercise.assessmentDueDate.isAfter(exercise.dueDate)
```

Comment:
I understand you just adapted this from the previous code, but I guess if statements would be more suitable and more readable here

Reply:
I just wanted to stick to the previous version :). i implemented a if/else chain now